src/compiler/emitter.ts

```diff
@@ -2072,20 +2072,19 @@ namespace ts {
 
         function emitSourceFile(node: SourceFile) {
             writeLine();
-            // If the node is a sourceFile and it has prologueDirective (statmentOffSet is not zero) that is synthesize
-            // We will need to emit detached comment here because emitPrologueDirective will not emit comments of such prologue directive
             const statements = node.statements;
-            // Emit detached comment if there iare no prologue directives or this is a synthesized prologue directives.
-            // The synthesized node will have no leading comment so some comments may be missed.
-            const shouldEmitDetachedComment = statements.length === 0 ||
-                !isPrologueDirective(statements[0]) ||
-                (isPrologueDirective(statements[0]) && nodeIsSynthesized(statements[0]));
-            if (emitBodyWithDetachedComments && shouldEmitDetachedComment) {
-                emitBodyWithDetachedComments(node, statements, emitSourceFileWorker);
-            }
-            else {
-                emitSourceFileWorker(node);
+            if (emitBodyWithDetachedComments) {
+                // Emit detached comment if there iare no prologue directives or this is a synthesized prologue directives.

```

Comment:
`// Emit detached comment if there are no prologue directives or if the first node is synthesized`.